### PR TITLE
Reduce duplication introduced in weaveworks/common migration

### LINKED ISSUE #343: Reduce duplication introduced in weaveworks/common migration
From https://github.com/grafana/dskit/pull/342/files/0e1e57319c8c2a483e6f4919a9946a81c017641b#r1283223263:

> `grpcutil.IsCancelled` is kinda duplicative of `grpcutil.IsGRPCContextCanceled`; do we need both?

And https://github.com/grafana/dskit/pull/342/files/0e1e57319c8c2a483e6f4919a9946a81c017641b#r1283233691:

> I think `github.com/grafana/dskit/errors.New` could be replaced by the Go standard library `errors.New`.

### PR DESCRIPTION
**Pull request by charleskorn**
**What this PR does**:

This PR fixes two instances of unnecessary duplication introduced in #342:

* `github.com/grafana/dskit/errors` is removed in favour of Go's `errors`
* `github.com/grafana/dskit/grpcutil.IsGRPCContextCanceled()` is removed in favour of `IsCanceled` in the same package.

As far as I can see, the first change has only minor impact outside this repository, and `grpcutil.IsGRPCContextCanceled()` is used in a single test in Mimir where the minor difference in behaviour is harmless.

**Which issue(s) this PR fixes**:

Fixes #343

**Checklist**
- [n/a] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: user/id.go [hunk 1]
```diff
@@ -19,14 +18,13 @@ const (
...
-	ErrDifferentOrgIDPresent = errors.Error("different org ID already present")
-	ErrTooManyOrgIDs         = errors.Error("multiple org IDs present")
+var (
+	ErrNoOrgID               = errors.New("no org id")
+	ErrDifferentOrgIDPresent = errors.New("different org ID already present")
+	ErrTooManyOrgIDs         = errors.New("multiple org IDs present")
 
-	ErrNoUserID               = errors.Error("no user id")
-	ErrDifferentUserIDPresent = errors.Error("different user ID already present")
-	ErrTooManyUserIDs         = errors.Error("multiple user IDs present")
```
Comment (aknuds1) [inline user/id.go:29]: Should this error be dropped? Why?

**Review by aknuds1 — COMMENTED** [review]
Just wondering about why an error is dropped.

File: user/id.go (hunk 1 above)
Comment (charleskorn) [reply user/id.go:29]: It's not used anywhere.

Comment (aknuds1) [reply user/id.go:29]: OK thanks for clarifying.

**Review by aknuds1 — APPROVED** [review]

**Pull request merged**